core/src/main/java/org/apache/carbondata/core/util/DeleteLoadFolders.java

```diff
@@ -128,7 +128,8 @@ private static void physicalFactAndMeasureMetadataDeletion(CarbonTable carbonTab
 
                 @Override public boolean accept(CarbonFile file) {
                   return (CarbonTablePath.isCarbonDataFile(file.getName()) ||
-                      CarbonTablePath.isCarbonIndexFile(file.getName()));
+                      CarbonTablePath.isCarbonIndexFile(file.getName()) ||
+                      CarbonTablePath.isCarbonMergeIndexFile(file.getName()));
```

Comment:
same may needs to be handled in org.apache.carbondata.core.metadata.SegmentFileStore#getSegmentFileForPhysicalDataPartitions

Please check

Reply:
actully for isIndexFile method, we are checking for startswith, so it considers merge index file also, as file extension is .carbonindexmerge. So this change is not required may be.

Reply:
OK got it